spec/ntheory_spec.rb

```diff
@@ -0,0 +1,82 @@
+require 'spec_helper'
+
+describe SymEngine do
+  before :each do
+  end
+
+  describe 'NTheory' do
+
+    describe '#gcd' do
+      context 'GCD of 2 and 4' do
+        it 'returns 2' do
+          f = SymEngine::gcd(2, 4)
+          expect(f).to eql(2)
+        end
+      end
+    end
+
+    describe '#lcm' do
+      context 'LCM of 2 and 4' do
+        it 'returns 4' do
+          f = SymEngine::lcm(2, 4)
+          expect(f).to eql(4)
+        end
+      end
+    end
+
+    describe '#nextprime' do
+      context 'NextPrime after 4' do
+        it 'returns 5' do
+          f = SymEngine::nextprime(4)
+          expect(f).to eql(5)
+        end
+      end
+    end
+
+    describe '#mod' do
+      context '5 mod 4' do
```

Comment:
Also add tests for other permutations of a positive and a negative number and see if they behave the same way as ruby, because I have heard they [do not behave the same way in python and C](http://stackoverflow.com/questions/1907565/c-python-different-behaviour-of-the-modulo-operation).